ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/DLModel.java

```diff
@@ -0,0 +1,307 @@
+                        Translator translator = getTranslator(engine, modelConfig);
+                        TranslatorFactory translatorFactory = getTranslatorFactory(engine, modelConfig);
+                        if (translatorFactory != null) {
+                            criteriaBuilder.optTranslatorFactory(translatorFactory);
+                        } else if (translator != null) {
+                            criteriaBuilder.optTranslator(translator);
+                        }
+
+                        Map<String, Object> arguments = getArguments(modelConfig);
+                        if (arguments != null && arguments.size() > 0) {
+                            for (Map.Entry<String,Object> entry : arguments.entrySet()) {
+                                criteriaBuilder.optArgument(entry.getKey(), entry.getValue());
+                            }
+                        }
+
+                        Criteria<Input, Output> criteria = criteriaBuilder.build();
+                        ZooModel<Input, Output> model = criteria.loadModel();
+                        Predictor<Input, Output> predictor = model.newPredictor();
+                        predictorList.add(predictor);
+                        modelList.add(model);
+
+                        // First request takes longer time. Predict once to warm up model.
+                        warmUp(predictor, modelId, modelConfig);
+                    }
+                    if (predictorList.size() > 0) {
+                        this.predictors = predictorList.toArray(new Predictor[0]);
+                        predictorList.clear();
+                    }
+                    if (modelList.size() > 0) {
+                        this.models = modelList.toArray(new ZooModel[0]);
+                        modelList.clear();
+                    }
+                    log.info("Model {} is successfully loaded on {} devices", modelId, devices.length);
+                    return null;
+                } catch (Throwable e) {
+                    String errorMessage = "Failed to load model " + modelId;
+                    log.error(errorMessage, e);
+                    close();
+                    if (predictorList.size() > 0) {
+                        closePredictors(predictorList.toArray(new Predictor[0]));
+                        predictorList.clear();
+                    }
+                    if (modelList.size() > 0) {
+                        closeModels(modelList.toArray(new ZooModel[0]));
+                        modelList.clear();
+                    }
+                    throw new MLException(errorMessage, e);
+                } finally {
+                    deleteFileQuietly(mlEngine.getLoadModelPath(modelId));
+                    Thread.currentThread().setContextClassLoader(contextClassLoader);
+                }
+            });
+        } catch (PrivilegedActionException e) {
+            String errorMsg = "Failed to load model";
```

Comment:
[minor] Can we please print the model id here also when you raise another PR?

Reply:
Fixed together with some failed UT